hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -195,14 +197,98 @@ def _exceptions_from_docstring(doc: str) -> Tuple[Type[Exception], ...]:
 # on analysis of type-annotated code to detect arguments which almost always
 # take values of a particular type.
 _GUESS_STRATEGIES_BY_NAME = (
+    (st.integers(0, 32), ["ndims"]),
+    (st.booleans(), ["keepdims"]),
     (st.text(), ["name", "filename", "fname"]),
     (st.floats(), ["real", "imag"]),
     (st.functions(), ["function", "func", "f"]),
+    (st.functions(returns=st.booleans(), pure=True), ["pred", "predicate"]),
     (st.iterables(st.integers()) | st.iterables(st.text()), ["iterable"]),
 )
 
 
-def _strategy_for(param: inspect.Parameter) -> Union[st.SearchStrategy, InferType]:
+def _type_from_doc_fragment(token: str) -> Optional[type]:
+    # Special cases for "integer" and for numpy array-like and dtype
+    if token == "integer":
+        return int
+    if "numpy" in sys.modules:
```

Comment:
No "float", akin to "integer"?

Reply:
Handled below, by

```python
    if "." not in token:
        return getattr(builtins, token, None)
```

"`integer`" is just a special case because there's a common natural-language name which isn't the name of the type.  We have similar special cases for "`sequence of`" or "`collection of`", but don't need it for "`float`".